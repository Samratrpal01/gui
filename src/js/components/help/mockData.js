export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
