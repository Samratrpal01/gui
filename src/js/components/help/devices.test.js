import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import Devices from './devices';
import { helpProps } from './mockData';

describe('GettingStarted Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<Devices {...helpProps} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
