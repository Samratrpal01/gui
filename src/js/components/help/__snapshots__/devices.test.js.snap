// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
}

<div
  class="flexbox column"
>
  <div
    class="one-column margin-bottom"
    style="flex-grow: 1; padding: 0px;"
  >
    <div
      class="margin-right-large"
    >
      <div
        class="flexbox center-aligned margin-bottom"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
          data-testid="LockOutlinedIcon"
          focusable="false"
          style="color: rgb(123, 123, 123);"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
        <div
          class="bold margin-left-small"
        >
          Static group
        </div>
      </div>
      <p
        class="help-content"
      >
        Select specific devices to add them to a static group. Filter to find devices in the 'All devices' UI, check the devices you want to add, and click the action button to assign them to a group.
      </p>
    </div>
    <img
      src="static-group-creation.gif"
      style="max-width: 100%;"
    />
  </div>
  <div
    class="one-column lightestgrey"
    style="flex-grow: 1; padding: 25px;"
  >
    <div
      class="margin-right-large"
    >
      <div
        class="flexbox center-aligned margin-bottom"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
          data-testid="AutorenewIcon"
          focusable="false"
          style="color: rgb(123, 123, 123);"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 6v3l4-4-4-4v3c-4.42 0-8 3.58-8 8 0 1.57.46 3.03 1.24 4.26L6.7 14.8c-.45-.83-.7-1.79-.7-2.8 0-3.31 2.69-6 6-6zm6.76 1.74L17.3 9.2c.44.84.7 1.79.7 2.8 0 3.31-2.69 6-6 6v-3l-4 4 4 4v-3c4.42 0 8-3.58 8-8 0-1.57-.46-3.03-1.24-4.26z"
          />
        </svg>
        <div
          class="bold margin-left-small margin-right"
        >
          Dynamic group
        </div>
        <span
          class="info uppercased"
          style="margin: 0px;"
        >
          Enterprise
        </span>
      </div>
      <p
        class="help-content"
      >
        You can set filters based on device attributes, and save them as a group. At any point in time, all devices that match the filters will be part of this group. This means that new devices will automatically join the group if they match the filters.
      </p>
      <p
        class="info"
      >
        Dynamic grouping is only available to Enterprise users. 
        <a
          href="mailto:contact@mender.io"
        >
          Contact us
        </a>
         to ask about upgrading.
      </p>
    </div>
    <img
      src="dynamic-group-creation.gif"
      style="max-width: 100%;"
    />
  </div>
</div>
`;
